tool/src/main/java/org/wildfly/security/tool/CredentialStoreCommand.java

```diff
@@ -292,6 +352,112 @@ public void execute(String[] args) throws Exception {
+                // prompt for passphrase
+                passphrase = prompt(false, ElytronToolMessages.msg.keyPassphrasePrompt(), true, ElytronToolMessages.msg.keyPassphrasePromptConfirm());
+            }
+            ElytronFilePasswordProvider passwordProvider = new ElytronFilePasswordProvider(createCredential(passphrase, PasswordCredential.class.getName()));
+
+            KeyPairCredential keyPairCredential;
+            String privateKeyContent;
+            String publicKeyContent = null;
+            String privateKeyString = cmdLine.getOptionValue(PRIVATE_KEY_STRING);
+            String publicKeyString = cmdLine.getOptionValue(PUBLIC_KEY_STRING);
+            String privateKeyFile = cmdLine.getOptionValue(PRIVATE_KEY_LOCATION_PARAM);
+            String publicKeyFile = cmdLine.getOptionValue(PUBLIC_KEY_LOCATION_PARAM);
+            String sshDirectory = cmdLine.getOptionValue(SSH_DIRECTORY_PARAM);
+            String knownHosts = cmdLine.getOptionValue(KNOWN_HOSTS_PARAM);
+
+            if (privateKeyFile != null) {
+                if (!Files.exists(Paths.get(privateKeyFile))) {
+                    setStatus(GENERAL_CONFIGURATION_ERROR);
+                    throw ElytronToolMessages.msg.keyFileDoesNotExist(privateKeyFile);
+                }
+                File keyFile = new File(privateKeyFile);
+                FileInputStream stream = new FileInputStream(keyFile);
+                byte[] keyData = new byte[stream.available()];
+                try {
+                    stream.read(keyData, 0, stream.available());
+                } finally {
+                    safeClose(stream);
+                }
+                privateKeyContent = new String(keyData);
+            } else if (privateKeyString != null) {
+                privateKeyContent = privateKeyString;
+            } else {
+                setStatus(GENERAL_CONFIGURATION_ERROR);
+                throw ElytronToolMessages.msg.noPrivateKeySpecified();
+            }
+
+            if (privateKeyContent.isEmpty()) {
+                setStatus(GENERAL_CONFIGURATION_ERROR);
+                throw ElytronToolMessages.msg.noPrivateKeySpecified();
+            }
+
+            if (publicKeyFile != null) {
+                if (!Files.exists(Paths.get(publicKeyFile))) {
+                    setStatus(GENERAL_CONFIGURATION_ERROR);
+                    throw ElytronToolMessages.msg.keyFileDoesNotExist(publicKeyFile);
+                }
+                File keyFile = new File(publicKeyFile);
+                FileInputStream stream = new FileInputStream(keyFile);
+                byte[] keyData = new byte[stream.available()];
+                try {
+                    stream.read(keyData, 0, stream.available());
+                } finally {
+                    safeClose(stream);
+                }
+                publicKeyContent = new String(keyData);
+            } else if (publicKeyString != null) {
+                publicKeyContent = publicKeyString;
+            }
```

Comment:
Should an exception be thrown here if no public key is specified?

Reply:
The exception should be thrown only if the keys are in PKCS8 format so I've made is so its thrown when its confirmed that the keys are in that format. Thanks!